question2/odbcdemo.c

```diff
@@ -0,0 +1,288 @@
+                    (SQLLEN *)&dataStruct3[0].length1);
+    if ((rc != SQL_SUCCESS) && (rc != SQL_SUCCESS_WITH_INFO)) {
+        printf("SQLBindCol(5) has Failed. RC=%d\n", rc);
+        ODBC_error(henv, hdbc, hstmt);
+        EnvClose(henv, hdbc);
+        exit(255);
+    }
+
+    // Set to bogus
+    dataStruct[0].length1 = -1;
+    dataStruct2[0].length1 = -1;
+    dataStruct3[0].length1 = -1;
+
+    int count = 0;
+
+    printf("Fetching result set...\n");
+    while (SQLFetch(hstmt) == SQL_SUCCESS) {
+        char *ptr;
+
+        if (count == 0) {
+            printf("LITERAL PREFIX\n");
+            printf("indicator(d): %d\n", dataStruct2[0].length1);
+            printf("indicator(lld): %lld\n", dataStruct2[0].length1);
+            printf("indicator(hex): ");
+
+            /* assign to char so we can dump out each byte */
+            ptr = (char *)&dataStruct2[0].length1;
+
+            for (i = 0; i < 8; i++) {
+                printf("%x", ptr[i]);
+            }
+            printf("\n");
+            printf("%-32s\n\n", dataStruct2[0].length1 == SQL_NULL_DATA
+                                    ? (UCHAR *)"NULL"
+                                    : dataStruct2[0].charCol1);
+
+            printf("LITERAL SUFFIX\n");
+            printf("indicator(d): %d\n", dataStruct3[0].length1);
+            printf("indicator(lld): %lld\n", dataStruct3[0].length1);
+            printf("indicator(hex): ");
+
+            /* assign to char so we can dump out each byte */
+            ptr = (char *)&dataStruct3[0].length1;
+
+            for (i = 0; i < 8; i++) {
+                printf("%x", ptr[i]);
+            }
+            printf("\n");
+            printf("%-32s\n\n", dataStruct3[0].length1 == SQL_NULL_DATA
+                                    ? (UCHAR *)"NULL"
+                                    : dataStruct3[0].charCol1);
+        }
+
+        printf("indicator(d): %d\n", dataStruct[0].length1);
+        printf("indicator(lld): %lld\n", dataStruct[0].length1);
+        printf("indicator(hex): ");
+
+        /* assign to char so we can dump out each byte */
+        ptr = (char *)&dataStruct[0].length1;
+
+        for (i = 0; i < 8; i++) {
+            printf("%x", ptr[i]);
+        }
+        printf("\n");
+        printf("%-32s\n\n", dataStruct[0].length1 == SQL_NULL_DATA
+                                ? (UCHAR *)"NULL"
+                                : dataStruct[0].charCol1);
+
+        dataStruct[0].length1 = -1;
+        dataStruct2[0].length1 = -1;
+        dataStruct3[0].length1 = -1;
+
+        count++;
+    }
+    if (rc == SQL_NO_DATA_FOUND) {
+        printf("SQLFetch returns: SQL_NO_DATA_FOUND\n");
+        goto end;
+    }
+    if ((rc != SQL_SUCCESS) && (rc != SQL_SUCCESS_WITH_INFO)) {
+        printf("SQLFetch has Failed. RC=%d\n", rc);
+        ODBC_error(henv, hdbc, hstmt);
+        goto end;
+    }
+/*
+** Free Bind Buffers
+*/
+end:
+    rc = SQLFreeStmt(hstmt, SQL_UNBIND);
```

Comment:
it is recommended to use SQLFreeHandle instead of SQLFreeStmt as of ODBC 3.X
ref: https://learn.microsoft.com/en-us/sql/relational-databases/native-client-odbc-api/sqlfreestmt?view=sql-server-ver15
```suggestion
    SQLFreeHandle(SQL_HANDLE_STMT, hstmt);
```